src/site/_filters/urls.js

```diff
@@ -14,18 +14,22 @@
  * limitations under the License.
  */
 const path = require('path');
+const {supportedLocales} = require('../../../shared/locale');
+const {findByUrl} = require('./find-by-url');
+
+const i18nRegex = /i18n\/\w+\//;
 
 /**
  * Removes any i18n data from a URL to determine the default URL.
- *
  * @param {string} url URL with possible i18n folder and locale in prefix.
  * @returns {string} URL without any i18n information.
  */
-const getDefaultUrl = (url) =>
-  url.replace(
-    /^\/i18n\/([a-z]{2}|[A-Z]{2}|ALL)(_([a-z]{2}|[A-Z]{2}|ALL))?\//,
-    '/',
-  );
+const getDefaultUrl = (url) => {
+  if (typeof url !== 'string') {
+    return url; // shows up for `permalink: false`
+  }
+  return url.startsWith('/i18n/') ? url.replace(i18nRegex, '') : url;
```

Comment:
Do we care if it's not a valid ISO code?

Reply:
This is more firebase-specific language combination code than iso code. I'll add a comment.